central/sensor/service/pipeline/clustermetrics/update_maxima.go

```diff
@@ -14,15 +14,40 @@ type BillingMetrics struct {
 }
 
 var (
+	// lastKnown stores the last known metrics per cluster.
+	lastKnown = map[string]BillingMetrics{}
```

Comment:
Why did you use a global variable here? I feel like this should be encapsulated into a cache entity together with the max maps.

Reply:
Yes, it could be encapsulated. Refactored in new PR.